docs/source/jit.rst

```diff
@@ -100,6 +100,129 @@ Example::
 
     my_script_module = torch.jit.script(MyScriptModule())
 
+Migrating to PyTorch 1.2 Recursive Scripting API
+------------------------------------------------
+This section details the changes to TorchScript in PyTorch 1.2. If you are new to TorchScript you can
+skip this section.
+
+Overview
+~~~~~~~~
+PyTorch 1.2 brings changes to the scripting API to close the usability gap between the TorchScript compiler and the tracer. Rather than inherit directly from ``torch.jit.ScriptModule``, a ``torch.nn.Module`` instance can be passed to ``torch.jit.script()``, which returns a compiled ``ScriptModule`` version of the module.
```

Comment:
I think this should be higher level (and doesn't need to talk about usability gaps). This is what I had for this section:

There are two main changes to the TorchScript API with 1.2.

1. `torch.jit.script` will now attempt to recursively compile functions,
   methods, and classes that it encounters. Once you call `torch.jit.script`,
   compilation is "opt-out", rather than "opt-in".
2. `torch.jit.script(nn_module_obj)` is now the preferred way to create
   `ScriptModule`s, instead of inheriting from `torch.jit.ScriptModule`.

These changes combine to provide a simpler, easier-to-use API for converting
your `nn.Module`s into `ScriptModule`s, ready to be optimzied and executed in a
non-Python environment. The new usage looks like this:

```
class Model(torch.nn.Module):
    def __init__(self):
        super(Model, self).__init__()
        self.conv1 = nn.Conv2d(1, 20, 5)
        self.conv2 = nn.Conv2d(20, 20, 5)

    def forward(self, x):
       x = F.relu(self.conv1(x))
       return F.relu(self.conv2(x))

my_model = Model()
my_scripted_model = torch.jit.script(my_model)
```